src/renderer/remote.ts

```diff
@@ -311,7 +311,7 @@ function metaToError (meta: { type: 'error', value: any, members: ObjectMember[]
 
 function handleMessage (channel: string, handler: Function) {
   ipcRenderer.on(channel, (event, passedContextId, id, ...args) => {
-    if (event.senderId !== 0) {
+    if (event.senderId !== 0 && event.senderId !== undefined) {
```

Comment:
I don't think this fully addresses the issue; with 28.x we see:

```
remote on git:main ❯ yarn install                                                      5:00PM
yarn install v1.22.19
[1/4] 🔍  Resolving packages...
[2/4] 🚚  Fetching packages...
[3/4] 🔗  Linking dependencies...
[4/4] 🔨  Building fresh packages...
success Saved lockfile.
$ tsc
src/renderer/remote.ts:314:15 - error TS2551: Property 'senderId' does not exist on type 'IpcRendererEvent'. Did you mean 'sender'?

314     if (event.senderId !== 0) {
                  ~~~~~~~~

  node_modules/electron/electron.d.ts:6002:5
    6002     sender: IpcRenderer;
             ~~~~~~
    'sender' is declared here.

src/renderer/remote.ts:315:81 - error TS2551: Property 'senderId' does not exist on type 'IpcRendererEvent'. Did you mean 'sender'?

315       console.error(`Message ${channel} sent by unexpected WebContents (${event.senderId})`);
                                                                                    ~~~~~~~~

  node_modules/electron/electron.d.ts:6002:5
    6002     sender: IpcRenderer;
             ~~~~~~
    'sender' is declared here.
```

Reply:
Yes, `senderId` has been removed, so it can only be checked for backward compatibility with older Electron versions.

If the build fails with Electron 28, can it be worked around by using `event['senderId']` instead? Or should we simply remove the gatekeeper completely?

Reply:
I'd be in favor of using `event['senderId']` for forward compatibility, personally.